analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R

```diff
@@ -111,28 +111,67 @@ hgg_lesions_df <- readr::read_tsv(
   )
 )
 
+# Read in the JSON file that contains the strings we'll use to include or
+# exclude samples for subtyping - see 00-HGG-select-pathology-dx
+path_dx_list <- jsonlite::fromJSON(
+  file.path(subset_dir, 
+            "hgg_subtyping_path_dx_strings.json")
+)
+
 #### Filter HGG defining lesions data.frame ------------------------------------
 
 # Filter the output file from `01-HGG-molecular-subtyping-defining-lesions.Rmd`
-# for samples classified as HGAT or those with defining lesions
-hgg_lesions_df <- hgg_lesions_df %>%
-  dplyr::filter(
-    short_histology == "HGAT" |
-      grepl("H3 G35 mutant|H3 K28 mutant", disease_type_reclassified)
-  )
+# for samples with defining lesions - there is a logical column 
+# `defining_lesion`
+hgg_lesions_df <- hgg_lesions_df %>% 
+  filter(defining_lesion)
 
 #### Filter metadata -----------------------------------------------------------
 
-# Filter metadata for HGAT and samples that should be classified
-# as High-grade glioma based on defining lesions
-hgg_metadata_df <- metadata %>%
-  dplyr::filter(
-    short_histology == "HGAT" |
-      sample_id %in% hgg_lesions_df$sample_id,
+# Filter metadata based on pathology diagnosis fields and include samples that 
+# should be classified as high-grade glioma based on defining lesions
+
+# First, filter to exclude cell lines
+tumor_metadata_df <- metadata %>%
+  filter(
     sample_type == "Tumor",
     composition == "Solid Tissue"
   )
 
+# Samples included on the basis of the pathology diagnosis fields
+
+path_dx_df <- tumor_metadata_df %>%
+  # Inclusion on the basis of strings in pathology_diagnosis and 
+  # pathology_free_text_diagnosis
+  filter(str_detect(pathology_diagnosis, 
+                    paste0(path_dx_list$include_path_dx, collapse = "|")) | 
+           str_detect(pathology_free_text_diagnosis, 
```

Comment:
These should be all lower case anyway, but maybe make it robust?
```suggestion
           str_detect(str_to_lower(pathology_free_text_diagnosis), 
```